app/controllers/otp_controller.rb

```diff
@@ -0,0 +1,17 @@
+class OtpController < ApplicationController
+  skip_before_action :authenticate_request, only: [ :send_otp ]
+  before_action :set_user, only: [ :send_otp ]
+  def send_otp
+    if @user
```

Comment:
in set_user you are throwing an error if user not found, and again you are checking @user in send_otp. it is required ?